### PR TITLE
Add note about TypeScript requiring types for namespaced attributes

### PR DESCRIPTION
**Pull request by titoBouzout**
- [x ] I have read the [Contribution guide](https://github.com/solidjs/solid-docs/blob/main/CONTRIBUTING.md)
- [ ] This PR references an issue (except for typos, broken links, or other minor problems)

### Description(required)

Someone made a note on discord that for example `prop:scrollTop` gives a type error, and asked if that was a bug.  Related docs page https://docs.solidjs.com/reference/jsx-attributes/prop

I understood the expectation, given that there is no indication that the typings for such a prop must be provided by the user. I understand we do not want to include types on each page and have a special page for types instead. So I'm including a note on each special namespaced attributes page indicating that the types must be provided and linking to the examples given in the TypeScript page

I'm not that sure about the format of the note as in "Note: text", or could be a callout but maybe It's too much, maybe we can bold the "Note" part? 

### Related issues & labels

- Closes #
- Suggested label(s) (optional):

### CODE REVIEW

File: src/routes/reference/jsx-attributes/bool.mdx [hunk 1]
```diff
@@ -19,3 +19,5 @@ This attribute is most useful for Web Components.
 <my-element />
 
 ```
+
+Note: Type definitions are required when using TypeScript. See the [TypeScript](/configuration/typescript#forcing-properties-and-custom-attributes) page for examples.
```
Comment (atilafassina) [inline src/routes/reference/jsx-attributes/bool.mdx]: You can use the Callout component

```ts
<Callout type="info" title="Strong-Typing Custom Properties">...</Callout>
```

File: src/routes/reference/jsx-attributes/prop.mdx [hunk 2]
```diff
@@ -8,3 +8,5 @@ Forces the prop to be treated as a property instead of an attribute.
 ```tsx
 <div prop:scrollTop={props.scrollPos + "px"} />
 ```
+
+Note: Type definitions are required when using TypeScript. See the [TypeScript](/configuration/typescript#forcing-properties-and-custom-attributes) page for examples.
```
Comment (atilafassina) [inline src/routes/reference/jsx-attributes/prop.mdx]: You can use the Callout component

```ts
<Callout type="info" title="Strong-Typing Custom Properties">...</Callout>
```

**Review by atilafassina — CHANGES_REQUESTED** [review]
Small nits about using the built components for highlighting/callout

**Comment by titoBouzout** [comment]
Thanks @atilafassina, changes made, using a `Callout` for the "Note:". Should be ready for review again

**Review by LadyBluenotes — APPROVED** [review]

**Review by atilafassina — APPROVED** [review]

**Pull request merged**